### PR TITLE
refactor(buttons): refactor buttons’ markup and styling - INNO-307

### PR DESCRIPTION
**Pull request by yhuard**
Closes #86.

Basically, I've imported @emeryro's work and rewritten the template (one template to rule them all). I still need to work on the "variants" (fractal-ly speaking), to provide more concrete examples.

TODO: 

- [x] Add more examples (using fractal's variants): icons, block, etc
- [x] Provide reference screenshots

### CODE REVIEW

File: framework/components/ecl-buttons/buttons/README.md [hunk 1]
```diff
@@ -1,42 +0,0 @@
...
-The submit and abort buttons are used to confirm a choice or send information. The abort button ('cancel') is optional.
-Style
-See Primary button.
-
-## Button block
-The button block is a series of default buttons with a fixed width. They link to child pages.
-Style
-See Default button style
-
-# Guidance
```
Comment (kalinchernev) [inline framework/components/ecl-buttons/buttons/README.md]: A bit painful to see that we already know it's unfinished work and someone will have to update urgently at worst moment

File: framework/components/ecl-breadcrumbs/_breadcrumbs.scss [hunk 2]
```diff
@@ -69,6 +67,10 @@ a.breadcrumb__link {
   // the breadcrumb segment width.
   width: 11px;
   color: #fff;
+
+  &::before {
```
Comment (kalinchernev) [inline framework/components/ecl-breadcrumbs/_breadcrumbs.scss]: breadcrumbs europa component library

Is as visible now, the arrow will have to be adjusted a bit.

File: framework/components/ecl-buttons/_buttons.scss [hunk 3]
```diff
@@ -1,354 +1,161 @@
-/*
-Buttons
+/**
+ * Buttons component
+ * @define button
```
Comment (kalinchernev) [inline framework/components/ecl-buttons/_buttons.scss]: Isn't this `@define ecl-button`?

File: framework/components/ecl-buttons/ecl-buttons.twig [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{% set type = type|default('button') %}
+{% set modifier = modifier|default('default') %}
+{% set extraClass = '' %}
+
+{% if isBlock is defined and isBlock == true %}
+  {% set extraClass = ' ecl-button--block' %}
+{% endif %}
+
+{% if icon is defined %}
+  {% set extraClass = extraClass ~ ' ecl-button--' ~ icon %}
```
Comment (kalinchernev) [inline framework/components/ecl-buttons/ecl-buttons.twig]: Adding a comment on this one will be appreciated, it currently seems a bit magical and takes time to understand.

File: framework/components/ecl-buttons/_buttons.scss [hunk 5]
```diff
@@ -1,354 +1,161 @@
-/*
-Buttons
```
Comment (kalinchernev) [inline framework/components/ecl-buttons/_buttons.scss]: Pointer cursor missing on hover, not sure if this is desired behavior.

buttons_hover

Comment (yhuard) [inline framework/components/ecl-buttons/_buttons.scss]: [W3C](https://www.w3.org/TR/CSS21/ui.html#cursor-props): "pointer" The cursor is a pointer that indicates a link.

File: framework/components/ecl-buttons/_buttons.scss (hunk 3 above)
Comment (yhuard) [inline framework/components/ecl-buttons/_buttons.scss]: Nope, the prefix should not be added in the definition :)

File: framework/components/ecl-buttons/ecl-buttons.twig (hunk 4 above)
Comment (yhuard) [inline framework/components/ecl-buttons/ecl-buttons.twig]: It definitely needs more documentation

File: framework/components/ecl-breadcrumbs/_breadcrumbs.scss (hunk 2 above)
Comment (yhuard) [inline framework/components/ecl-breadcrumbs/_breadcrumbs.scss]: Will see when working on breadcrumbs 👼

File: framework/components/ecl-buttons/_buttons.scss (hunk 5 above)
Comment (kalinchernev) [inline framework/components/ecl-buttons/_buttons.scss]: in my face :D

**Pull request merged**